internal/configuration/configuration.go

```diff
@@ -69,7 +69,13 @@ func NewConfigurationManager(dataDir string) *Manager {
 	return &mgr
 }
 
-func (m *Manager) RegisterObserver(observer Observer) {
+// RegisterObserver registers an Observer to the config, that will get called
+// when any config changes happens.
+// If withInitConfig is set, observer will be called with the current config.
+func (m *Manager) RegisterObserver(observer Observer, withInitConfig bool) {
+	if withInitConfig {
```

Comment:
I was thinking on a a slightly different way to implement this:
The `Observer` interface can be extended with `Init` method that will be responsible to making sure the observer is set properly (it will be each observer responsibility if to call Update from Init or perform any initialization logic or do nothing).
```
type Observer interface {
	Init(configuration models.DeviceConfigurationMessage) error
	Update(configuration models.DeviceConfigurationMessage) error
}
```
Then the main will remain mostly the same, without a need to specify for each observer how it should be started.
In other words, the responsibility to init the observer will be done within each observer.

Reply:
Makes a lot of sense to me, I'll do in this way. Thanks @masayag